### PR TITLE
feat(promise): stdin.promise

### PR DESCRIPTION
**Pull request by ahmadnassri**
ES6 compatible promises ... 

requires `node >= 0.11` (let me know if you prefer `0.10` compatibility)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,3 +49,14 @@ module.exports.buffer = function (cb) {
 		cb(Buffer.concat(ret, len));
 	});
 };
+
+module.exports.promise = function () {
+	return new Promise(function (_resolve, _reject) {
```
Comment (kevva) [inline index.js]: Why the `_`?

**Comment by kevva** [comment]
Yes, 0.10 compatibility is needed.

File: index.js (hunk 1 above)
Comment (ahmadnassri) [inline index.js]: just a personal preference, I'll update for this PR.

**Comment by ahmadnassri** [comment]
> Yes, 0.10 compatibility is needed.

any prefernce on a Promise polyfill? something lightweight, I've used [jakearchibald/es6-promise](https://github.com/jakearchibald/es6-promise) in the past.

**Comment by ahmadnassri** [comment]
ended up going with `bluebird` ... easy enought to change if needed.

**Comment by kevva** [comment]
https://github.com/floatdrop/pinkie-promise

**Comment by sindresorhus** [comment]
Thanks, but I don't want to add promise cruft this way. When I'm ready to adopt promises I'll switch completely over. That time is not yet here though.

I don't think trying to get all the modules you use to adopt promises is feasable. Instead just use the `promisify` method in your preferred promise module: https://github.com/petkaantonov/bluebird/blob/master/API.md#promisification

**Comment by ahmadnassri** [comment]
@kevva `pinkie-promise` is awesome, thanks!

@sindresorhus that's alright, curious though, what's you're reasoning for not adopting promises just yet? (even if not fully).

> I don't think trying to get all the modules you use to adopt promises is feasible.

very true, and I tend to avoid over complicating already complicated modules that way, this one seemed small and simple enough that adding a promise method would have little impact on complexity.

**Comment by sindresorhus** [comment]
> what's you're reasoning for not adopting promises just yet?

I realize promises can be useful, but the overhead of adding them with polyfills and all that does IMHO not justify its minor improvement in usability. Especially not for tiny modules like this one. I will switch when I can target promises without a polyfill or any conditional.

I have adopted promises on [one module](https://github.com/sindresorhus/got).

**Comment by sindresorhus** [comment]
Ok, changed my mind. I completely converted this module to use promises. Enjoy! I didn't realize until now that Node.js 0.12 has promise support. I went with no polyfill as this module is mostly used with CLI tools where requiring Node.js 0.12 is not a problem.

**Comment by ahmadnassri** [comment]
ha! thanks :)

**Pull request closed without merging**